guides/hack/21-XHP/01-setup.md

```diff
@@ -1,13 +1,41 @@
-## Namespace Support
-
-Support for namespaced XHP classes (elements) is enabled by default since
+Support for XHP and namespaced XHP classes (elements like `<p>`) is enabled by default since
 HHVM 4.73. It comes with many related
 [XHP changes](https://hhvm.com/blog/2020/09/02/XHP-namespaces-and-syntax.html)
 compared to previous versions.
 
-We recommend using this version or newer, since it's more thoroughly
-tested and doesn't require any extra configuration, but XHP namespace support
-can also be enabled in older HHVM versions (since around HHVM 4.46) by adding
+## The XHP-Lib Library
+
+While the XHP syntax is part of Hack, a large part of the implementation is in a
+library called [XHP-Lib](https://github.com/hhvm/xhp-lib/) that needs to be
+installed via [Composer](https://getcomposer.org/) (add `facebook/xhp-lib` to your `composer.json` manually
+or by running `composer require facebook/xhp-lib ^4.0` or `^3.0`).
+
+This includes the base classes and interfaces, and definitions of standard HTML elements.
+
+### XHP-Lib Version
+
+There are currently two major supported versions of XHP-Lib:
+
+- **XHP-Lib v4:** to be used when XHP namespace support is enabled. Declares all
+  base classes, interfaces and elements in namespaces (e.g. standard HTML
+  elements are in `Facebook\XHP\HTML`). It is also more strict (e.g. disallows
+  most mutations after an element is rendered) and deprecates some features
+  (e.g. attribute "clobbering" in `XHPHTMLHelpers`).
+- **XHP-Lib v3:** to be used in older HHVM versions or when XHP namespace
+  support is disabled. Declares everything in the global namespace, with the
+  exception of `Facebook\XHP\ChildValidation`.
+
+All the following guides are written with the assumption that XHP namespace
+support is enabled and XHP-Lib v4 is used, but there are notes pointing out any
+major differences&mdash;look for **Historical note** sections.
+
+<span class="fbOnly fbIcon">XHP namespaces are not enabled in Facebook's WWW
+repository, so all **Historical note** sections apply.</span>
+
+## Enable XHP for older HHVM versions
```

Comment:
The heading isn't quite accurate here. This switch does not enable/disable XHP. It enables the `xhp class` rules, which makes xhp classes interoperate with namespaces.

Reply:
With mangling enabled, you declare classes in the v3/fb manner, using a colon prefix to indicate xhp-ness. This mangling turns `class :fb:hello-world` into `class xhp_fb__hello_world`. Xhp depends on the fact that this is `\xhp_fb__hello_world` and not `\SomeNamespace\xhp_fb__hello_world`. With mangling disabled (the default), you declare classes in the v4 manner, using the xhp class modifier. `xhp class fb:hello_world` becomes `CurrentNamespace\fb\hello_world`. Xhp no longer depends on the fact that the names get dumped into the root namespace. By changing this setting, you change the parsing rules. You are not disabling xhp (xhp is always enabled). In mangling mode `function f(:fb:block $b): void;` becomes `f(xhp_fb__block $b): void;`. In no-mangling mode, it become `function f(\CurrentNamespace\fb\block $b): void;`.

Reply:
Changing to "Enable XHP Class Rules" and "Disable XHP Class Rules" respectively.